### PR TITLE
fix: fixed onChange being fired twice on clear input

### LINKED ISSUE #18027: [Bug]: Controllled combobox fires onchange twice when input field is cleared
### Package

@carbon/react

### Browser

Chrome

### Package version

v1.70.0

### React version

_No response_

### Description

When I click the x in the combo box input field to clear the content, the onchange event is fired twice. The first time it fires with a selectedItem value of null which is expected. However then it fires again with a selectedItem value of the previous selection. It should not be called a 2nd time.
This is likely related to https://github.com/carbon-design-system/carbon/issues/17847 which was recently fixed.

### Reproduction/example

https://stackblitz.com/edit/github-k8ejla?file=src%2FApp.jsx&preset=node=

### Steps to reproduce

Run the slackblitz sample with devtools open so you can see the console logs.
Select an item on the ComboBox.
Click the x to clear the input field
Problem: The onChange function is called twice. You can see the debug log printed 2 times to the console with the different selectedItem values

Image

### Suggested Severity

None

### Application/PAL

IBM Cloud IAM-UI

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by guidari**
Closes #18027

Added a new validayion inside the if clause to aovid two `onChanges` to be called when the clear button is clicked.

#### Testing / Reviewing

- Test the Combobox clear functionality with the console open.

- [ ] Delete console.log

### CODE REVIEW

File: packages/react/src/components/ComboBox/ComboBox.stories.js [hunk 1]
```diff
@@ -219,6 +219,7 @@ export const _fullyControlled = () => {
   ];
   const [value, setValue] = useState(options[0]);
   const onChange = ({ selectedItem }) => {
+    console.log('onChange', selectedItem);
```
Comment (preetibansalui) [inline packages/react/src/components/ComboBox/ComboBox.stories.js:222]: Do we need this console?

**Review by annawen1 — APPROVED** [review]
LGTM!

**Review by 2nikhiltom — APPROVED** [review]
LGTM

**Comment by guidari** [comment]
Duplicate fix from https://github.com/carbon-design-system/carbon/pull/18041#pullrequestreview-2434031494

**Pull request closed without merging**